### ISSUE TITLE
Dashed line with 'vector' source displayed incorrectly at some zoom levels

### ISSUE THREAD

**Issue by newlinefu** [maplibre/maplibre-gl-js#2448]
At certain positions of the dashed line and map zoom level, the dashed line becomes unexpected on one of the segments. Problem not reproducible when using GeoJSON source with the same line coordinates

**maplibre-gl-js version**:
1.5.2 (also reproduced on 2.4.0 and 3.0.0-pre.5)
**browser**:
Chrome 112.0.5615.138
### Steps to Trigger Behavior

1. Add vector source to map. Geometry of expected line:

```json
[
      [
        15.613515457854128,
        50.912416032719435
      ],
      [
        15.897903442382812,
        52.71549904109216
      ],
      [
        17.40054623512618,
        52.65364171608446
      ],
      [
        17.40054623512618,
        51.65364171608446
      ],
      [
        16.92718505859375,
        51.300569443062926
      ],
      [
        16.40054623512618,
        51.65364171608446
      ]
    ]
```
2.  Add styles for layer with paint

```
{
    'line-dasharray': [20, 8],
    'line-color': 'black',
    'line-width': 1.5
}
```

3. Change zoom of map

### Link to Demonstration
Unfortunately, the backend part is needed for the demonstration. [Here](https://github.com/newlinefu/maplibre-dash-demonstration) you can take a small go application with an [entry point](https://github.com/newlinefu/maplibre-dash-demonstration/blob/main/cmd/map-view/main.go) and [html file](https://github.com/newlinefu/maplibre-dash-demonstration/blob/main/web/index.html)

![maplibre-dash-demonstration](https://user-images.githubusercontent.com/63716478/234563563-0953a951-fa0b-4428-ba15-f8f549db2662.gif)

### Expected Behavior
The dash of line is stable at any zoom level

### Actual Behavior
Dash changing at some zoom levels

**Comment by HarelM** [comment]
Is it possible for you to create a renderv test that demonstrates this behavior maybe?

**Comment by HarelM** [comment]
Wow! Amaizing! The only thing left is to fix it 😀

**Comment by HarelM** [comment]
Thanks for fixing it!! @newlinefu 
Resolved by #2479
